compiler/parser/parser.peg

```diff
@@ -578,19 +578,19 @@ ShapeOp
     }
 
 JoinOp
-  = style:JoinStyle JOIN rightInput:JoinRightInput _ e:JoinExpr optArgs:(_ FlexAssignments)? {
+  = style:JoinStyle JOIN alias:OptJoinAlias rightInput:JoinRightInput _ e:JoinExpr {
```

Comment:
I think the alias and the condition should be adjacent because

```
join (
  from ...
) as {l,r} where l.a=r.b
```

reads better and is clearer than

```
join as {l,r} (
  from ...
) on l.a=r.b
```

Reply:
I think agree but this from the doc written by @mccanne so let's hear what he has to say